ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/table/LoadTable.java

```diff
@@ -270,10 +269,9 @@ static TableLocationTuple tableLocation(ImportTableDesc tblDesc, Database parent
     Path dataPath = fromURI;
     Path tmpPath = tgtPath;
 
-    // if move optimization is enabled, copy the files directly to the target path. No need to create the staging dir.
+    // if acid tables, copy the files directly to the target path. No need to create the staging dir.
     LoadFileType loadFileType;
-    if (replicationSpec.isInReplicationScope() &&
-            context.hiveConf.getBoolVar(REPL_ENABLE_MOVE_OPTIMIZATION)) {
+    if (replicationSpec.isInReplicationScope() && AcidUtils.isTransactionalTable(table)) {
```

Comment:
For partitioned table, we always assume that it's an acid table case, shouldn't we assume the same here as well?

Reply:
This is only for the ptests which use non acid managed tables.